gson/src/main/java/com/google/gson/internal/sql/SqlDateTypeAdapter.java

```diff
@@ -62,7 +63,9 @@ public java.sql.Date read(JsonReader in) throws IOException {
     try {
       Date utilDate;
       synchronized (this) {
+        TimeZone originalTimeZone = format.getTimeZone(); // Save the original time zone
         utilDate = format.parse(s);
+        format.setTimeZone(originalTimeZone); // Restore the original time zone after parsing
```

Comment:
Please use `try-finally` here as well, to be safe (in case time zone was already changed despite an exception being thrown):
```suggestion
        TimeZone originalTimeZone = format.getTimeZone(); // Save the original time zone
        try {
          utilDate = format.parse(s);
        } finally {
          format.setTimeZone(originalTimeZone); // Restore the original time zone after parsing
        }
```
(you might to have to reformat the code above)